pkg/grpc/auth.go

```diff
@@ -0,0 +1,27 @@
+package grpc
+
+import (
+	"context"
+	"net/http"
+
+	v1 "github.com/dapr/dapr/pkg/messaging/v1"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/metadata"
+)
+
+func setAPIAuthenticationMiddlewareUnary(apiToken, authHeader string) grpc.UnaryServerInterceptor {
+	return func(ctx context.Context, req interface{}, info *grpc.UnaryServerInfo, handler grpc.UnaryHandler) (interface{}, error) {
+		md, _ := metadata.FromIncomingContext(ctx)
+		token := md.Get(authHeader)
```

Comment:
what happens if token is nil ?

Reply:
token is a slice of strings (not a ref), it can't be nil